src/ApparentHorizons/Expansion.hpp

```diff
@@ -0,0 +1,28 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
+
+/// \ingroup SurfacesGroup
+/// \brief Expansion of a `Strahlkorper`. Should be zero on apparent horizons.
+///
+/// \details Eq. (5) in https://arxiv.org/abs/gr-qc/9606010.
+/// `r_hat`, `radius`, `normal_one_form`, and `d2x_radius`
+/// depend on the shape of the Strahlkorper but not on the metric, and
+/// can be computed from a Strahlkorper using ComputeItems in
+/// `StrahlkorperTags`.  `upper_spatial_metric`, `extrinsic_curvature`, and
+/// `christoffel_2nd_kind` are 3-metric quantities that are independent
+/// of the Strahlkorper except they are evaluated on the grid points
+/// associated with the Strahlkorper.
+template <typename Frame>
+DataVector expansion(
```

Comment:
This function seems to do a lot of computation inside it. Are the computed quantities not useful elsewhere? If they are I think it would make more sense to have them split out

Reply:
A few of the simple ones (one_over_magnitude and surf_norm_vector) may be useful for observers, but not sure which ones.  Split them later, once we have observers and we know what we need?